src/common/acl_principal.c

```diff
@@ -309,3 +311,116 @@ daos_acl_principal_to_gid(const char *principal, gid_t *gid)
 	D_FREE(buf);
 	return rc;
 }
+
+static bool
+is_special_type(enum daos_acl_principal_type type)
+{
+	switch (type) {
+	case DAOS_ACL_OWNER:
+	case DAOS_ACL_OWNER_GROUP:
+	case DAOS_ACL_EVERYONE:
+		return true;
+	case DAOS_ACL_USER:
+	case DAOS_ACL_GROUP:
+	default:
+		break;
+	}
+
+	return false;
+}
+
+static int
+get_principal_type_from_str(const char *principal_str,
+			    enum daos_acl_principal_type *type)
+{
+	if (strncmp(principal_str, DAOS_ACL_PRINCIPAL_OWNER,
+		    DAOS_ACL_MAX_PRINCIPAL_BUF_LEN) == 0) {
+		*type = DAOS_ACL_OWNER;
+		return 0;
+	}
+
+	if (strncmp(principal_str, DAOS_ACL_PRINCIPAL_OWNER_GRP,
+		    DAOS_ACL_MAX_PRINCIPAL_BUF_LEN) == 0) {
+		*type = DAOS_ACL_OWNER_GROUP;
+		return 0;
+	}
+
+	if (strncmp(principal_str, DAOS_ACL_PRINCIPAL_EVERYONE,
+		    DAOS_ACL_MAX_PRINCIPAL_BUF_LEN) == 0) {
+		*type = DAOS_ACL_EVERYONE;
+		return 0;
+	}
+
+	/*
+	 * Named user or group will be designated by prefix
+	 */
+	if (strncmp(principal_str, USER_PREFIX, strlen(USER_PREFIX)) == 0) {
+		*type = DAOS_ACL_USER;
+		return 0;
+	}
+
+	if (strncmp(principal_str, GRP_PREFIX, strlen(GRP_PREFIX)) == 0) {
+		*type = DAOS_ACL_GROUP;
+		return 0;
+	}
+
+	return -DER_INVAL;
+}
+
+static const char *
+get_start_of_name(const char *principal_str, enum daos_acl_principal_type type)
+{
+	size_t idx;
+
+	if (type == DAOS_ACL_USER)
+		idx = strlen(USER_PREFIX);
```

Comment:
since USER_PREFIX and GRP_PREFIX are constants, you could use (sizeof(USER_PREFIX) - 1) which would be a compile time constant rather than calling strlen

Reply:
Good catch. Will fix.